### PR TITLE
Optimize quota calculation for propfind

### PR DESCRIPTION
**Pull request by icewind1991**
For #14173

cc @DeepDiver1975 @PVince81

### CODE REVIEW

File: lib/private/connector/sabre/directory.php [hunk 1]
```diff
@@ -31,6 +30,8 @@ class OC_Connector_Sabre_Directory extends OC_Connector_Sabre_Node
 	 */
 	private $dirContent;
 
+	private $quotaInfo;
```
Comment (DeepDiver1975) [inline lib/private/connector/sabre/directory.php]: PHPDoc please

**Comment by DeepDiver1975** [comment]
Please elaborate if unit test can be applied - prime rule for PRs - see https://github.com/owncloud/core/commit/b533aad21697de29e8d69d7bb99b0309f2114d07#diff-6a3371457528722a734f3c51d9238c13

**Comment by icewind1991** [comment]
Added a unit test for webdav's getQuotaInfo

**Comment by PVince81** [comment]
:+1: I checked with debug statements and can confirm that this reduces the call to cache's "get()" from 3 to 2 calls.

**Comment by MorrisJobke** [comment]
:+1:

**Comment by PVince81** [comment]
Regression found with `quota-used-space`: https://github.com/owncloud/core/issues/24320#issuecomment-216251388

**Pull request merged**